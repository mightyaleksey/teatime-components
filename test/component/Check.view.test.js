'use strict';

const Check = require('../../component/Check');
const React = require('react');
const renderer = require('react-test-renderer');

test('defaultProp #size should have a value `m`', () => {
  expect(Check.defaultProps.size).toBe('m');
});

test('Renders correctly with default props', () => {
  const tree = renderer.create(
    <Check
      name='control'>
      I Agree
    </Check>
  );

  expect(tree).toMatchSnapshot();
});

test('Renders correctly with defined props', () => {
  const tree = renderer.create(
    <Check
      checked
      className='outer'
      disabled
      hasLabel={false}
      id='controls-id'
      name='control'
      value='controls value'>
      Make Awesome!
    </Check>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
