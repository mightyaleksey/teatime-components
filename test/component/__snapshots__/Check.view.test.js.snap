// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with default props 1`] = `
<span
  className="wrapper"
  onChange={undefined}
>
  <input
    checked={undefined}
    className="native"
    defaultChecked={undefined}
    disabled={undefined}
    id="_teatime1"
    name="control"
    onBlur={undefined}
    onChange={[Function]}
    onFocus={undefined}
    type="checkbox"
    value={undefined}
  />
  <label
    className="control"
    htmlFor="_teatime1"
  />
  <label
    className="label"
    htmlFor="_teatime1"
  >
    I Agree
  </label>
</span>
`;

exports[`Renders correctly with defined props 1`] = `
<span
  className="wrapper outer"
  onChange={undefined}
>
  <input
    checked={true}
    className="native"
    defaultChecked={undefined}
    disabled={true}
    id="controls-id"
    name="control"
    onBlur={undefined}
    onChange={[Function]}
    onFocus={undefined}
    type="checkbox"
    value="controls value"
  />
  <label
    className="control"
    htmlFor="controls-id"
  >
    Make Awesome!
  </label>
</span>
`;
